### DISCUSSION TITLE
Difference between get_diffuse and get_diffuse_half?

### DISCUSSION THREAD

**Discussion by eballeste** [bnpr/Malt#205] [Q&A] [answered]
I am trying to learn about NPR and material properties in general and just learned about the differences between diffuse, specular, emission and ambient colors. Looking through how shaders work in MALT, I noticed that most GLSL examples use the `get_diffuse_half` method when creating the diffuse vector. 

Besides the obvious assumptions I can make from the name of the method itself, what is the reasoning behind the preference for `get_diffuse_half` and what helps determine when to reach for `get_diffuse` vs `get_diffuse_half`?

**Comment by pragma37** [comment, marked as answer]
It's just a style choice, use whatever you prefer:
https://developer.valvesoftware.com/wiki/Half_Lambert
https://technology.riotgames.com/news/valorant-shaders-and-gameplay-clarity

**Reply by eballeste** [reply]
I don't have a preference as of yet, I'm just trying to learn, thanks for the links, they explained a lot!
